packages/angular_devkit/build_angular/test/browser/web-worker_spec_large.ts

```diff
@@ -0,0 +1,183 @@
+        );
+        // worker bundle contains worker code.
+        expect(workerContent).toContain('hello from worker');
+        expect(workerContent).toContain('bar');
+
+        const mainContent = virtualFs.fileBufferToString(
+          host.scopedSync().read(join(outputPath, 'main.js')),
+        );
+        // main bundle references worker.
+        expect(mainContent).toContain('0.worker.js');
+      }),
+      // Doesn't show any warnings.
+      tap(() => expect(logger.includes('WARNING')).toBe(false, 'Should show no warnings.')),
+    ).toPromise().then(done, done.fail);
+  });
+
+  it('minimizes and hashes worker', (done) => {
+    host.writeMultipleFiles(workerFiles);
+    const overrides = {
+      experimentalWebWorkerTsConfig: 'src/tsconfig.worker.json',
+      outputHashing: 'all',
+      optimization: true,
+    };
+    runTargetSpec(host, browserTargetSpec, overrides).pipe(
+      tap((buildEvent) => expect(buildEvent.success).toBe(true)),
+      tap(() => {
+        const workerBundle = host.fileMatchExists(outputPath,
+          /0\.[0-9a-f]{20}\.worker\.js/) as string;
+        expect(workerBundle).toBeTruthy('workerBundle should exist');
+        const workerContent = virtualFs.fileBufferToString(
+          host.scopedSync().read(join(outputPath, workerBundle)),
+        );
+        expect(workerContent).toContain('hello from worker');
+        expect(workerContent).toContain('bar');
+        expect(workerContent).toContain('"hello"===t&&postMessage(o.foo)');
+
+        const mainBundle = host.fileMatchExists(outputPath, /main\.[0-9a-f]{20}\.js/) as string;
+        expect(mainBundle).toBeTruthy('mainBundle should exist');
+        const mainContent = virtualFs.fileBufferToString(
+          host.scopedSync().read(join(outputPath, mainBundle)),
+        );
+        expect(mainContent).toContain(workerBundle);
+      }),
+    ).toPromise().then(done, done.fail);
+  });
+
+  it('rebuilds TS worker', (done) => {
+    host.writeMultipleFiles(workerFiles);
+    const overrides = {
+      experimentalWebWorkerTsConfig: 'src/tsconfig.worker.json',
+      watch: true,
+    };
+
+    let buildCount = 0;
+    let phase = 1;
+    const workerPath = join(outputPath, '0.worker.js');
+    let workerContent = '';
+
+    runTargetSpec(host, browserTargetSpec, overrides, DefaultTimeout * 3).pipe(
+      // Note(filipesilva): Wait for files to be written to disk... and something else?
+      // 1s should be enough for files to be written to disk.
+      // However, with a 1s to 3s delay, I sometimes (roughly 1 in 3 tests) saw TS compilation
+      // succeeding and emitting updated content, but the TS loader never called for
+      // 'src/worker/dep.ts'.
+      // But increasing this delay to 5s lead to no failed test in over 40 runs locally.
+      // In CI I still saw a rebuild fail with 5s, so I increased it to 10s.
+      // I think there might be a race condition related to child compilers somewhere in webpack.
```

Comment:
Still saw a failure with the 10s debounce time in https://circleci.com/gh/angular/angular-cli/43023#tests/containers/1. Need to debug further.

Reply:
@filipesilva Is it possible the child compiler needs additional plugins? If it does and their configuration is the same, adding a string match pattern to the [`plugins[]` option](https://github.com/GoogleChromeLabs/worker-plugin#plugins) in worker-plugin would copy them over.

I'm not sure where to look in this repo to peek at how file watching is being done, any tips appreciated!

Reply:
I'm looking at this currently. I'm not 100% sure of what the problem is but I've managed to reproduce it locally. I think it's related to the shared bits of our TS compilation pipeline.

Reply:
I think I figured it out, let's continue the conversation in the other comment (https://github.com/angular/angular-cli/pull/13700#discussion_r268188296).